packages/flutter_tools/gradle/flutter.gradle

```diff
@@ -397,12 +397,41 @@ class FlutterPlugin implements Plugin<Project> {
               "io.flutter:flutter_embedding_$flutterBuildMode:$engineVersion"
             )
         }
+
         // Wait until the Android plugin loaded.
         pluginProject.afterEvaluate {
+            if (pluginProject.android.compileSdkVersion > project.android.compileSdkVersion) {
+                project.logger.quiet("Warning: The plugin ${pluginName} requires Android SDK version ${pluginProject.android.compileSdkVersion.substring(8)}.")
+            }
             project.android.buildTypes.all addEmbeddingDependencyToPlugin
         }
     }
 
+     /** Prints error message and fix for any plugin compileSdkVersion that are higher than the project. */
+    private void detectLowCompileSdkVersion() {
+        project.afterEvaluate {
+            int projectCompileSdkVersion = project.android.compileSdkVersion.substring(8) as int
+            int maxPluginCompileSdkVersion = projectCompileSdkVersion
+            int numProcessedPlugins = getPluginList().size()
+
+            getPluginList().each { plugin ->
+                Project pluginProject = project.rootProject.findProject(plugin.key)
+                pluginProject.afterEvaluate {
+                    int pluginCompileSdkVersion = pluginProject.android.compileSdkVersion.substring(8) as int
```

Comment:
nit add: `assert pluginProject.android.compileSdkVersion.startsWith('android-')` to ensure that this invariant is true prior to `substr(8)`.  who knows if this changes to `android-foo-` in the future.